src/Input/Input.js

```diff
@@ -299,7 +299,7 @@ export type Props = {
    *
    * @param {object} event The event source of the callback
    */
-  onChange?: (event: SyntheticInputEvent<>) => void,
+  onChange?: (event: SyntheticInputEvent<HTMLInputElement>) => void,
```

Comment:
This could be a TextArea if `multiline` is `true`. `Input` also allows for an arbitrary `InputComponent` prop that would allow this to be either a subtype of `HTMLElement` or a `React.Component`. 

The correct thing to do here might to do a dynamic type refinement.

Reply:
Ahh, I was under the impression that the `Input` component mapped to `input` elements.. If that's not the case, I'm happy to change this to an `HTMLElement`.

Reply:
Actually, it'd be nice for `onChange` handlers to use `.value`. How do you feel about using `SyntheticInputEvent<HTMLInputElement|HTMLTextAreaElement>`?

Reply:
The problem here is that this component allows `InputComponent` to be anything. `HTMLInputElement|HTMLTextAreaElement` covers most cases, realistically, but won't be workable for those other cases.

@oliviertassinari any insight into why this would allow anything other than `input` and `textarea` types? Are there any real world reasons this would be, say, a `span`?

Reply:
> Are there any real world reasons this would be, say, a span?

It can be a `select` and a `div` in the case of the Select.

Reply:
Ok cool. @dhui would it be reasonable for your use case to make this an `HTMLElement`, and then do a runtime refinement when you need something more specific? Ex:

```
if (event.target instanceof HTMLInputElement) {
  ...
```

Reply:
That works for me. I'll push a new commit that changes this to a `HTMLElement`